src/components/editor/ocr/ocr-editor-component.ts

```diff
@@ -0,0 +1,278 @@
+import * as html from '@hyperapp/html'
```

Comment:
I think I mentioned it before, but I feel like it should be mentioned again here. The idea behind OCR, originally, was to provide an offline[1] means of extracting text from a handwritten document (image) to facilitate searching as you can search a textual note.

The OCR panel here serves a different purpose. That's not to say it's not good or bad, it's just a different purpose. This is fine for drawing a character or word from another language and performing OCR so you don't have to look it up in a reference table to be able to insert it into the document, but other than that it doesn't aid a user trying to write down length text.

So, I feel this isn't heading towards the intended goal and should at the very least be extracted into a separate branch from the primary handwriting components. Following that, the extended goal, as agreed upon, is to provide offline[1] parsing to enable searching across handwritten documents. Continuing down this line of a OCR panel for inputting single words or characters isn't going to aid in completing the agreed upon goals.

offline: To carry out asynchronously from the rest of the application -- after the data has been saved, some time later, at a periodic time perhaps. 

@KelvinKKLin

Reply:
Agreed. Can the OCR be extracted to provide both user-initiated translation of notes as it is currently doing, as well as search-initiated translation of notes? Is the OCR module fast enough for this? If not, can we consider caching the translated hand-written notes somewhere?